internal/unionstore/pipelined_memdb.go

```diff
@@ -0,0 +1,286 @@
+		metrics.TiKVPipelinedFlushLenHistogram.Observe(float64(p.flushingMemBuffer.Len()))
+		metrics.TiKVPipelinedFlushSizeHistogram.Observe(float64(p.flushingMemBuffer.Size()))
+		err := p.flushFunc(p.flushingMemBuffer)
+		p.onFlushing.Store(false)
+		// Send the error to errCh after onFlushing status is set to false.
+		// this guarantees the onFlushing.Store(true) in another goroutine's MayFlush happens after onFlushing.Store(false) in this function.
+		p.errCh <- err
+	}()
+	return true, nil
+}
+
+// FlushWait will wait for all flushing tasks are done and return the error if there is a failure.
+func (p *PipelinedMemDB) FlushWait() error {
+	if p.flushingMemBuffer != nil {
+		err := <-p.errCh
+		// cleanup the flushingMemBuffer so the next call of FlushWait will not wait for the error channel.
+		p.flushingMemBuffer = nil
+		return err
+	}
+	return nil
+}
+
+// Iter implements the Retriever interface.
+func (p *PipelinedMemDB) Iter([]byte, []byte) (Iterator, error) {
+	return nil, errors.New("pipelined memdb does not support Iter")
+}
+
+// IterReverse implements the Retriever interface.
+func (p *PipelinedMemDB) IterReverse([]byte, []byte) (Iterator, error) {
+	return nil, errors.New("pipelined memdb does not support IterReverse")
+}
+
+// SetEntrySizeLimit sets the size limit for each entry and total buffer.
+func (p *PipelinedMemDB) SetEntrySizeLimit(entryLimit, bufferLimit uint64) {
+	p.memBuffer.SetEntrySizeLimit(entryLimit, bufferLimit)
+}
+
+// InspectStage implements MemBuffer interface.
+func (p *PipelinedMemDB) InspectStage(int, func([]byte, kv.KeyFlags, []byte)) {}
+
+// SnapshotIter implements MemBuffer interface.
+func (p *PipelinedMemDB) SnapshotIter(k, upperBound []byte) Iterator {
+	// TODO: return an iterator which output error.
+	return nil
+}
+
+// SnapshotIterReverse implements MemBuffer interface.
+func (p *PipelinedMemDB) SnapshotIterReverse(k, lowerBound []byte) Iterator {
+	return nil
+}
+
+// SnapshotGetter implements MemBuffer interface.
+func (p *PipelinedMemDB) SnapshotGetter() Getter {
+	return nil
+}
+
+func (p *PipelinedMemDB) Len() int {
+	return p.memBuffer.Len() + p.len
+}
+
+func (p *PipelinedMemDB) Size() int {
+	return p.memBuffer.Size() + p.size
+}
+
+func (p *PipelinedMemDB) OnFlushing() bool {
+	return p.onFlushing.Load()
+}
+
+// SetMemoryFootprintChangeHook sets the hook for memory footprint change.
+func (p *PipelinedMemDB) SetMemoryFootprintChangeHook(hook func(uint64)) {}
+
+// Mem returns the memory usage of MemBuffer.
+func (p *PipelinedMemDB) Mem() uint64 {
+	return 0
+}
+
+// Reset the PipelinedMemDB to init state.
+func (p *PipelinedMemDB) Reset() {
+	p.memBuffer.Reset()
+	p.flushingMemBuffer = nil
+	p.len, p.size = 0, 0
+	p.onFlushing.Store(false)
+	select {
+	case <-p.errCh:
+	default:
+	}
+}
+
+// Staging is not supported for PipelinedMemDB, it returns 0 handle.
+func (p *PipelinedMemDB) Staging() int {
+	return 0
+}
+
+// Cleanup implements MemBuffer interface.
```

Comment:
Need to comment the belowing interfaces are not supported by the `PipelinedMemDB` as they do not return errors.

Reply:
Let's just panic.

Reply:
Done.